lib/queries.js

```diff
@@ -119,68 +182,443 @@ export async function updateOvoNumberAndUri(ovoStructuredIdUri, wegwijsOvo) {
 
           FILTER (?ovoNotation IN ("OVO-nummer"@nl, "OVO-nummer"))
         }
-      }
-    }
-  `;
-
+       }
+     }
+   
+   `;
   const result = await query(queryStr);
 
   if (result.results.bindings.length) {
     const bindings = result.results.bindings;
-    return bindings.map(binding => {
+    return bindings.map((binding) => {
       return {
-        kbo: binding.kbo.value,
+        kbo: binding.kbo?.value,
         kboStructuredId: binding.kboStructuredId.value,
         ovo: binding.ovo?.value,
         ovoStructuredId: binding.ovoStructuredId?.value,
-      }
+        kboOrg: binding.kboOrg?.value,
+        abbOrg: binding.abbOrg?.value,
+        kboId: binding.kboId?.value,
+        changeTime: binding.changeTime?.value,
+      };
     });
   }
   return null;
 }
 
+export async function createNewKboOrg(kboOrg, kboId) {
+  const newKboOrgIdUuId = uuid();
+  const newKboIdUri = `http://data.lblod.info/id/kboBestuurseenheden/${newKboOrgIdUuId}`;
+
+  const contactPointUuidId = uuid();
+  const contactPointUri = `http://data.lblod.info/id/contact-punten/${contactPointUuidId}`;
+
+  const adressUuidId = uuid();
+  const adressUri = `http://data.lblod.info/id/adressen/${adressUuidId}`;
+
+  let updateStr = `
+  INSERT {
+    GRAPH ?g {
+      ${sparqlEscapeUri(adressUri)} a <http://www.w3.org/ns/locn#Address> ;
+                                      <http://mu.semte.ch/vocabularies/core/uuid> ${sparqlEscapeString(
+                                        adressUuidId
+                                      )} `;
+  if (kboOrg.formattedAddress) {
+    updateStr += `;
+       <http://www.w3.org/ns/locn#fullAddress> ${sparqlEscapeString(
+         kboOrg.formattedAddress
+       )}`;
+  }
+  updateStr += " .";
+
+  updateStr += `                               
+      ${sparqlEscapeUri(contactPointUri)} a <http://schema.org/ContactPoint> ;
+                                            <http://mu.semte.ch/vocabularies/core/uuid> ${sparqlEscapeString(
+                                              contactPointUuidId
+                                            )} `;
+  if (kboOrg.email) {
+    updateStr += `; <http://schema.org/email> ${sparqlEscapeString(
+      kboOrg.email
+    )}`;
+  }
+  if (kboOrg.phone) {
+    updateStr += `; <http://schema.org/telephone> ${sparqlEscapeString(
+      kboOrg.phone
+    )}`;
+  }
+  if (kboOrg.website) {
+    updateStr += `; <http://xmlns.com/foaf/0.1/page> ${sparqlEscapeString(
+      kboOrg.website
+    )}`;
+  }
+  updateStr += `
+    ; <http://schema.org/contactType> ${sparqlEscapeString("Kbocontact")} ;
+    <http://www.w3.org/ns/locn#address> ${sparqlEscapeUri(adressUri)}.`;
+
+  updateStr += `    ${sparqlEscapeUri(
+    newKboIdUri
+  )} a <http://mu.semte.ch/vocabularies/ext/KboEenheid> ;
+                                        <http://mu.semte.ch/vocabularies/core/uuid> ${sparqlEscapeString(
```

Comment:
**Issue:** The `dct:source` predicate is missing: `<dct:source> <https://economie.fgov.be/> `

Reply:
change applied